compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5721,6 +5668,90 @@ private BType resolveQueryType(SymbolEnv env, BLangExpression selectExp, BType c
         }
     }
 
+    void solveSelectTypeAndResolveType(BLangQueryExpr queryExpr, BLangExpression selectExp, BType type,
+                                       BType collectionType, List<BType> selectTypes, List<BType> resolvedTypes,
+                                       SymbolEnv env, AnalyzerData data, boolean isReadonly,
```

Comment:
I think it does not need to pass this variable `isReadonly`. 
We can define it as a local variable here and set the correct value when it is needed.

Reply:
But this method will recursively call when the type is BIntersectionType.

Reply:
Yes, ignore this